airflow/www_rbac/security.py

```diff
@@ -181,13 +181,21 @@ def init_role(self, role_name, role_vms, role_perms):
         if not role:
             role = self.add_role(role_name)
 
-        role_pvms = []
-        for pvm in pvms:
-            if pvm.view_menu.name in role_vms and pvm.permission.name in role_perms:
-                role_pvms.append(pvm)
-        role.permissions = list(set(role_pvms))
-        self.get_session.merge(role)
-        self.get_session.commit()
+        if len(role.permissions) == 0:
+            logging.info('Initializing permissions for role:%s in the database.', role_name)
+            role_pvms = []
+            for pvm in pvms:
+                if pvm.view_menu.name in role_vms and pvm.permission.name in role_perms:
+                    role_pvms.append(pvm)
+            role.permissions = list(set(role_pvms))
+            self.get_session.merge(role)
+            self.get_session.commit()
+        else:
+            logging.info('Existing permissions for the role:%s within the database will persist.', role_name)
+            for pvm in pvms:
+                if pvm not in role.permissions:
+                    if pvm.view_menu.name in role_vms and pvm.permission.name in role_perms:
+                        role.permissions.append(pvm)
```

Comment:
Glad to see the test passes!

Just wondering why the `else` chunk works without session merge/commit.

Reply:
may be append takes care of it. But anyways reverted these changes because if default permissions are removed in the UI for a specific role, it will get added back when init_role is called because of which the permission view changes will not persist. Instead, fixed the test by adding the missing permission.